File: composeApp/src/androidMain/kotlin/com/ivangarzab/kluvs/ui/clubs/GeneralTab.kt
package com.ivangarzab.kluvs.ui.clubs

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.MoreVert
import androidx.compose.material3.Card
import androidx.compose.material3.CardDefaults
import androidx.compose.material3.DropdownMenu
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.HorizontalDivider
import androidx.compose.material3.Icon
import androidx.compose.material3.IconButton
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.tooling.preview.PreviewLightDark
import androidx.compose.ui.unit.dp
import com.ivangarzab.kluvs.R
import com.ivangarzab.kluvs.clubs.presentation.BookInfo
import com.ivangarzab.kluvs.clubs.presentation.ClubDetails
import com.ivangarzab.kluvs.clubs.presentation.DiscussionInfo
import com.ivangarzab.kluvs.model.Role
import com.ivangarzab.kluvs.theme.KluvsTheme
import com.ivangarzab.kluvs.ui.components.NextDiscussionCard
import com.ivangarzab.kluvs.ui.components.NoSectionData
import com.ivangarzab.kluvs.ui.components.NoTabData

@Composable
fun GeneralTab(
    modifier: Modifier = Modifier,
    clubDetails: ClubDetails? = null,
    userRole: Role? = null,
    onEditClub: () -> Unit = {},
    onDeleteClub: () -> Unit = {},
) {
    if (clubDetails == null) {
        NoTabData(
            modifier = modifier,
            text = R.string.no_club_details
        )
        return
    }

    Column(
        modifier = modifier,
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        // Club Info Card
        Card(
            modifier = Modifier.fillMaxWidth(),
            colors = CardDefaults.cardColors(
                containerColor = MaterialTheme.colorScheme.surface
            )
        ) {
            Column(modifier = Modifier.padding(16.dp)) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.CenterVertically
                ) {
                    Column(modifier = Modifier.weight(1f)) {
                        Text(
                            text = stringResource(
                                R.string.founded_in_x,
                                clubDetails.foundedYear ?: stringResource(R.string.na)
                            ),
                            color = MaterialTheme.colorScheme.onSurfaceVariant,
                            style = MaterialTheme.typography.bodyMedium,
                        )
                        Spacer(Modifier.height(4.dp))
                        Text(
                            text = stringResource(R.string.x_members, clubDetails.memberCount),
                            color = MaterialTheme.colorScheme.onSurfaceVariant,
                            style = MaterialTheme.typography.bodyMedium
                        )
                    }

                    if (userRole == Role.OWNER) {
                        ClubOverflowMenu(
                            onEdit = onEditClub,
                            onDelete = onDeleteClub
                        )
                    }
                }
            }
        }

        HorizontalDivider(color = MaterialTheme.colorScheme.surfaceVariant)

        // Current Book Card
        Card(
            modifier = Modifier.fillMaxWidth(),
            colors = CardDefaults.cardColors(
                containerColor = MaterialTheme.colorScheme.surface
            )
        ) {
            Column(modifier = Modifier.padding(16.dp)) {
                Text(
                    text = stringResource(R.string.current_book),
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    style = MaterialTheme.typography.titleMedium
                )
                Spacer(Modifier.height(8.dp))
                clubDetails.currentBook?.let { clubInfo ->
                    Text(
                        text = clubInfo.title,
                        color = MaterialTheme.colorScheme.onSurface,
                        style = MaterialTheme.typography.bodyLarge
                    )
                    Spacer(Modifier.height(4.dp))
                    Text(
                        text = clubInfo.author,
                        color = MaterialTheme.colorScheme.onSurfaceVariant,
                        style = MaterialTheme.typography.bodyMedium
                    )
                    Spacer(Modifier.height(4.dp))
                    Row(horizontalArrangement = Arrangement.spacedBy(4.dp)) {
                        Text(
                            text = clubDetails.currentBook?.year ?: stringResource(R.string.na),
                            color = MaterialTheme.colorScheme.onSurfaceVariant,
                            style = MaterialTheme.typography.bodyMedium
                        )
                        clubDetails.currentBook?.pageCount?.let { pages ->
                            Text(
                                text = stringResource(R.string.vertical_separator),
                                color = MaterialTheme.colorScheme.onSurfaceVariant,
                                style = MaterialTheme.typography.bodyMedium
                            )
                            Text(
                                text = stringResource(R.string.x_pages, pages),
                                color = MaterialTheme.colorScheme.onSurfaceVariant,
                                style = MaterialTheme.typography.bodyMedium
                            )
                        }
                    }
                } ?: NoSectionData(text = R.string.no_book_data)
            }
        }

        HorizontalDivider(color = MaterialTheme.colorScheme.surfaceVariant)

        // Next Discussion Card
        Card(
            modifier = Modifier
                .fillMaxWidth(),
            colors = CardDefaults.cardColors(
                containerColor = MaterialTheme.colorScheme.surface
            )
        ) {
            Column(modifier = Modifier.padding(16.dp)) {
                Text(
                    text = stringResource(R.string.next_discussion),
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    style = MaterialTheme.typography.titleMedium
                )
                Spacer(Modifier.height(8.dp))

                clubDetails.nextDiscussion?.let { discussion ->
                    NextDiscussionCard(
                        modifier = Modifier.fillMaxWidth(),
                        title = discussion.title,
                        location = discussion.location,
                        formattedDate = discussion.formattedDate,
                    )
                } ?: NoSectionData(text = R.string.no_upcoming_discussion)
            }
        }
    }
}

@Composable
private fun ClubOverflowMenu(
    onEdit: () -> Unit,
    onDelete: () -> Unit,
) {
    var expanded by remember { mutableStateOf(false) }

    Box {
        IconButton(onClick = { expanded = true }) {
            Icon(
                imageVector = Icons.Default.MoreVert,
                contentDescription = "Club options",
                tint = MaterialTheme.colorScheme.onSurfaceVariant
            )
        }
        DropdownMenu(
            expanded = expanded,
            onDismissRequest = { expanded = false }
        ) {
            DropdownMenuItem(
                text = { Text("Edit") },
                onClick = {
                    expanded = false
                    onEdit()
                }
            )
            DropdownMenuItem(
                text = {
                    Text(
                        text = "Delete",
                        color = MaterialTheme.colorScheme.error
                    )
                },
                onClick = {
                    expanded = false
                    onDelete()
                }
            )
        }
    }
}

@PreviewLightDark
@Composable
fun Preview_GeneralTab() = KluvsTheme {
    GeneralTab(
        modifier = Modifier
            .background(color = MaterialTheme.colorScheme.surface)
            .fillMaxSize()
            .padding(horizontal = 16.dp, vertical = 8.dp),
        clubDetails = ClubDetails(
            clubId = "club1",
            clubName = "Test Club Name",
            memberCount = 6,
            foundedYear = "2026",
            currentBook = BookInfo(
                title = "1984",
                author = "George Orwell",
                year = "1948",
                pageCount = 169
            ),
            nextDiscussion = DiscussionInfo(
                title = "Discussion #1",
                location = "Discord",
                formattedDate = "Tomorrow at 7:00 PM"
            )
        ),
        userRole = Role.OWNER
    )
}

File: iosApp/iosApp/Views/Clubs/GeneralTab.swift
import SwiftUI
import Shared

struct GeneralTab: View {
    let clubDetails: Shared.ClubDetails?
    var userRole: Shared.Role? = nil
    var onEditClub: () -> Void = {}
    var onDeleteClub: () -> Void = {}

    private var isOwner: Bool { userRole == .owner }

    var body: some View {
        ScrollView {
            if let clubDetails = clubDetails {
                VStack(spacing: 12) {
                    // Club Info Card
                    VStack(alignment: .leading, spacing: 8) {
                        if isOwner {
                            HStack {
                                Spacer()
                                Menu {
                                    Button("Edit Club Name") { onEditClub() }
                                    Button("Delete Club", role: .destructive) { onDeleteClub() }
                                } label: {
                                    Image(systemName: "ellipsis.circle")
                                        .foregroundColor(.secondary)
                                }
                            }
                        }

                        if let foundedYear = clubDetails.foundedYear {
                            Text(String(format: NSLocalizedString("label_founded_in", comment: ""), foundedYear))
                                .font(.subheadline)
                                .foregroundColor(.secondary)
                        }

                        Text(String(format: NSLocalizedString("label_members_count", comment: ""), clubDetails.memberCount))
                            .font(.subheadline)
                            .foregroundColor(.secondary)
                    }
                    .frame(maxWidth: .infinity, alignment: .leading)
                    .padding()

                    Divider()

                    // Current Book Card
                    VStack(alignment: .leading, spacing: 8) {
                        Text("section_current_book")
                            .font(.headline)
                            .foregroundColor(.secondary)

                        if let book = clubDetails.currentBook {
                            Text(book.title)
                                .font(.body)
                                .fontWeight(.medium)

                            Text(book.author)
                                .font(.subheadline)
                                .foregroundColor(.secondary)

                            HStack(spacing: 4) {
                                Text(book.year ?? String(localized: "label_not_available"))
                                    .font(.subheadline)
                                    .foregroundColor(.secondary)

                                if let pageCount = book.pageCount {
                                    Text("•")
                                        .foregroundColor(.secondary)
                                    Text(String(format: NSLocalizedString("label_pages", comment: ""), pageCount.int32Value))
                                        .font(.subheadline)
                                        .foregroundColor(.secondary)
                                }
                            }
                        } else {
                            NoSectionData(text: String(localized: "empty_no_book_data"))
                        }
                    }
                    .frame(maxWidth: .infinity, alignment: .leading)
                    .padding()

                    Divider()

                    // Next Discussion Card
                    VStack(alignment: .leading, spacing: 8) {
                        Text("section_next_discussion")
                            .font(.headline)
                            .foregroundColor(.secondary)

                        if let discussion = clubDetails.nextDiscussion {
                            NextDiscussionCard(
                                title: discussion.title,
                                location: discussion.location,
                                formattedDate: discussion.formattedDate
                            )
                        } else {
                            NoSectionData(text: String(localized: "empty_no_upcoming_discussion"))
                        }
                    }
                    .frame(maxWidth: .infinity, alignment: .leading)
                    .padding()
                }
                .padding()
            } else {
                NoTabData(text: String(localized: "empty_no_club_details"))
            }
        }
    }
}

#Preview {
    GeneralTab(clubDetails: nil)
}
